lib/tasks/journeys_report.rake

```diff
@@ -2,60 +2,85 @@ require 'csv'
 require 'json'
 require 'aws-sdk-s3'
 
+def get_last_run(bucket_name)
+  s3 = Aws::S3::Resource.new
+  # Amazon S3 lists objects in alphabetical order
+  objects = s3.bucket(bucket_name).objects().collect(&:key)
+  timestamp = objects.any? ? objects.last.split('/').first : '0'
+  last_run_date = DateTime.strptime(timestamp, '%s')
+  last_run_date
+end
+
 namespace :journeys do
   desc 'Exports a JSON report of all journeys'
   task json_report: :environment do
     timestamp = DateTime.now.utc.to_i
+    bucket_name = ENV['S3_REPORTING_BUCKET_NAME'] || raise('S3_REPORTING_BUCKET_NAME not defined')
+
+    if ENV['RAILS_ENV'].to_s == 'development' || ENV['RAILS_ENV'].to_s == ''
+      Aws.config.update(
+        endpoint: 'http://localhost:4572',
+        credentials: Aws::Credentials.new('fakeid', 'fakesecret'),
+        region: 'eu-west-2',
+        force_path_style: true,
+      )
+    end
+
+    last_report_date = get_last_run(bucket_name)
+    puts last_report_date
 
     Supplier.order(:key).each do |supplier|
       filename = "#{supplier.key}-#{timestamp}.json"
       puts "Generating #{filename}..."
       moves = []
       Move
           .where("ID in (select move_id from journeys WHERE supplier_id = '#{supplier.id}')")
+          .where("moves.updated_at > '#{last_report_date}'")
+          .where('moves.updated_at < now()')
           .order('moves.date ASC, moves.reference ASC').each do |move|
         journeys = []
         Journey.where(move: move).default_order.each do |journey|
           journeys << {
-            timestamp: journey.client_timestamp.strftime('%d/%m %H:%M:%S'),
-            from: {
-              title: journey.from_location.title,
-              id: journey.from_location.id,
-              nomis_agency_id: journey.from_location.nomis_agency_id,
-            },
-            to: {
-              title: journey.to_location.title,
-              id: journey.to_location.id,
-              nomis_agency_id: journey.to_location.nomis_agency_id,
-            },
+            id: journey.id,
+            supplier: supplier.name,
+            timestamp: journey.client_timestamp.iso8601,
+            from: journey.from_location.nomis_agency_id,
+            to: journey.to_location.nomis_agency_id,
```

Comment:
I've flattened the `to`and `from`to a string as we only seem to require the nomis_agency_id.